sys/src/cmd/screenlock.c

```diff
@@ -211,62 +211,63 @@ void
 lockscreen(void)
 {
 	enum { Nfld = 5, Fldlen = 12, Cursorlen = 2*4 + 2*2*16, };
-	char *s;
-	char buf[Nfld*Fldlen], *flds[Nfld], newcmd[128], cbuf[Cursorlen];
-	int fd, dx, dy;
-	Image *i;
-	Point p;
-	Rectangle r;
-	Tm *tm;
-
-	fd = open("/dev/screen", OREAD);
+
+	int fd = open("/dev/screen", OREAD);
 	if(fd < 0)
 		error("can't open /dev/screen: %r");
+
+	char buf[Nfld*Fldlen], *flds[Nfld], newcmd[128];
 	if(read(fd, buf, Nfld*Fldlen) != Nfld*Fldlen)
 		error("can't read /dev/screen: %r");
 	close(fd);
+
 	buf[sizeof buf-1] = 0;
 	if(tokenize(buf, flds, Nfld) != Nfld)
 		error("can't tokenize /dev/screen header");
 	snprint(newcmd, sizeof newcmd, "-r %s %s %d %d",
 		flds[1], flds[2], atoi(flds[3]) - 1, atoi(flds[4]) - 1);
 	newwindow(newcmd);
-	if (initdraw(nil, nil, "screenlock") < 0)
+	if(initdraw(nil, nil, "screenlock") < 0)
 		sysfatal("initdraw failed");
 	if(display == nil)
 		error("no display");
 
 	/* screen is now open and covered.  grab mouse and hold on tight */
 	procrfork(grabmouse, nil, 4096, RFFDG);
 	procrfork(blanker, nil, 4096, RFFDG);
+
+	draw(screen, screen->r, display->black, nil, ZP);
```

Comment:
You could swap these two lines and do draw(screen, r, etc. etc.)
Not a big deal but ...